includes/admin/class-wcs-admin-meta-boxes.php

```diff
@@ -97,11 +97,15 @@ public function remove_meta_boxes() {
 	}
 
 	/**
-	 * Don't save some order related meta boxes
+	 * Don't save some order related meta boxes.
+	 *
+	 * @see woocommerce_process_shop_order_meta
+	 *
+	 * @param int $post_or_order_id
+	 * @param WC_Order|WP_Post $post_or_order_object
 	 */
-	public function remove_meta_box_save( $post_id, $post ) {
-
-		if ( 'shop_subscription' === $post->post_type ) {
+	public function remove_meta_box_save( $post_or_order_id, $post_or_order_object ) {
```

Comment:
```suggestion
	public function remove_meta_box_save( $subscription_id, $subscription ) {
```

Small nit-pick: I don't think we need to specify that this is a post or order in the variable name because when WC moves away entirely from WP Posts we will have to go back and rename these variables. I think we should stick to naming these variables what we expect them to be i.e. `$order_id` or `$subscription_id`, or just `$id` in a general sense and `$order`, `$subscription` or `$object`.

If we expect the object/id to be both an order and/or subscription we can just use `$order`.

Reply:
Ah, good point, thanks. I was preferring to use `$post_or_order_object` under the influence of the [HPOS upgrade guide](https://github.com/woocommerce/woocommerce/wiki/High-Performance-Order-Storage-Upgrade-Recipe-Book#how-to-enable-hpos), although I see that with many WC-specific hooks, `$order` will be provided.

Reply:
Addressed in bfa5da3 and d2d0694